### PR TITLE
Disable validation of enum values for incomplete model definitions

### PR DESCRIPTION
**Pull request by norberttech**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    <li>Disable validation of enum values for incomplete model definitions</li>
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>

### CODE REVIEW

**Comment by Tetsuya-Takiguchi** [comment]
Hi, @norberttech.

Unfortunately, I found a bug.
The class name you can get from `EventCode::class` is `AmazonPHP\SellingPartner\Model\FulfillmentOutbound\EventCode`.
However, the class name of the `$class` variable to be compared was `\AmazonPHP\SellingPartner\Model\FulfillmentOutbound\EventCode`.
Therefore, there is no `$class` value in the `$brokenModelDefinitions` array in the `in_array` function.
We would appreciate it if you could confirm this.

File: src/AmazonPHP/SellingPartner/ObjectSerializer.php [hunk 1]
```diff
@@ -342,6 +344,16 @@ public static function deserialize(Configuration $configuration, $data, string $
         }
 
         if (\method_exists($class, 'getAllowableEnumValues')) {
+            $brokenModelDefinitions = [
+                EventCode::class, // https://github.com/amazon-php/sp-api-sdk/issues/191
```
Comment (norberttech) [inline src/AmazonPHP/SellingPartner/ObjectSerializer.php:348]: @Tetsuya-Takiguchi here is the EventCode class, you are saying that $class variable is getting different value?

Comment (Tetsuya-Takiguchi) [reply src/AmazonPHP/SellingPartner/ObjectSerializer.php:348]: @norberttech 
Yes, I am. That is correct.
The value of the `$class` variable is `\AmazonPHP\SellingPartner\Model\FulfillmentOutbound\EventCode`, with a backslash at the beginning of the string.
In contrast, the value that can be retrieved from `EventCode::class` is `AmazonPHP\SellingPartner\Model\FulfillmentOutbound\EventCode`, with no backslash at the beginning of the string.

Comment (norberttech) [reply src/AmazonPHP/SellingPartner/ObjectSerializer.php:348]: oh now I see, thanks! I adjusted the logic to trim left backslash before comparison which should solve the problem. 
https://github.com/amazon-php/sp-api-sdk/pull/194

Comment (Tetsuya-Takiguchi) [reply src/AmazonPHP/SellingPartner/ObjectSerializer.php:348]: @norberttech 
Thank you for the correction.
I will pull #194 and check it out.

**Pull request merged**